recipe_scrapers/lekkerensimpel.py

```diff
@@ -29,10 +32,39 @@ def image(self):
         return image["content"] if image else None
 
     def ingredients(self):
-        return self.schema.ingredients()
+        if self.schema.ingredients():
+            return self.schema.ingredients()
+
+        ingredient_header = self.soup.find("strong", string="Benodigdheden:")
+        if ingredient_header:
+            return self.process_ingredients(
+                ingredient_header.parent.parent.find_next("ul")
+            )
+
+        ingredient_header = self.soup.find("p", string="Benodigdheden:")
+        if ingredient_header:
+            return self.process_ingredients(ingredient_header.parent.find_next("ul"))
```

Comment:
We don't have a best-practice for this at the moment, but perhaps we can try out some ideas (or at least discuss them) given the opportunity here.

I think what's happening in this case is that we have three layers of web-archaeology for the site: `schema.org` (`self.schema.ingredients`) is the most recent, `strong` tags are the second-most-recent, and `p` ingredients are the oldest we're aware of.

For now, perhaps a comment above each of the three code blocks to indicate something about the age of the content?  It doesn't have to be precise for now - `# approx Y2018` or similar would be fine.

Or alternatively: maybe there's a better way to annotate this, or maybe we shouldn't bother?  (personally I think it could help future maintainers to understand, but that's debatable too)

Reply:
Well, maybe we should allow multiple scrapers per website? So you can see that there are different strategies to scrape a single webpage? That would make the code easier to understand and more straight forward. Downside is, this would not allow combinations of strategies, like we can have them right now.  ingredients in a new format, with older instructions format. 

Another option could be to have separate methods per strategy in the same scraper? A comment could do the trick, but is less obvious, as I cannot tell you exactly when the formats are changed. It looks a bit like the site is manually edited. As there seems to be a mix somewhere in 2016. 

But I get where you are coming from.

Reply:
I guess abstractly I think of _most_ websites as having a single, structured format for any given point in time.

So then the _theoretical_ (non-reality, I emphasize) ideal could be to have timestamp ranges associated with each scraper.  The combination of `domain(url) + timestamp(url)` would then indicate what version of each scraper to use.  Figuring out the range limits would be an iterative process -- not too different to fixing the scrapers.

In that architecture, each scraper would be independent and unaware of the other temporal scraper versions for the same site.

In practice, though: I don't think we have the scale of contributors nor analytical capability -- at the moment, anyway -- to rearchitect the code to support that kind of usage.  Not to mention that it would require some amout of stored state (to figure out the appropriate timestamp for each URL).

There'd also no doubt be edge cases even with that kind of architecture (for example, sites that include pages that use two or more different formats at any given time).

I think the short summary is that keeping the code in a single scraper is probably sensible in this case, because most of methods are shared between both page formats, and we don't really have a widely-used way to select alternate scrapers within a single domain name...

(...although technically we do have one case where we do this: `allrecipes` can dynamically select a class to instantiate based on a URL pattern match)